packages/cli/src/commands/screenshot-diff/screenshot-diff.command.ts

```diff
@@ -18,199 +24,244 @@ import {
+            headScreenshot.height !== headScreenshot.height
+          ) {
+            return [
+              {
+                identifier,
+                outcome: "different-size",
+                headScreenshotFile: `screenshots/${screenshotFileName}`,
+                baseScreenshotFile: `screenshots/${screenshotFileName}`,
+                baseWidth: baseScreenshot.width,
+                baseHeight: baseScreenshot.height,
+                headWidth: headScreenshot.width,
+                headHeight: headScreenshot.height,
+              },
+            ];
+          }
 
-      totalMismatchPixels += comparisonResult.mismatchPixels;
+          const comparisonResult = compareImages({
+            base: baseScreenshot,
+            head: headScreenshot,
+            pixelThreshold: diffPixelThreshold,
+          });
 
-      logger.debug({
-        screenshotFileName,
-        mismatchPixels: comparisonResult.mismatchPixels,
-        mismatchFraction: comparisonResult.mismatchFraction,
-      });
+          await writeScreenshotDiff({
+            baseReplayId,
+            headReplayId,
+            screenshotFileName,
+            diff: comparisonResult.diff,
+          });
 
-      await writeScreenshotDiff({
-        baseReplayId,
-        headReplayId,
-        screenshotFileName,
-        diff: comparisonResult.diff,
-      });
+          return [
+            {
+              identifier,
+              outcome:
+                comparisonResult.mismatchFraction > diffThreshold
+                  ? "diff"
+                  : "no-diff",
+              headScreenshotFile: `screenshots/${screenshotFileName}`,
+              baseScreenshotFile: `screenshots/${screenshotFileName}`,
+              width: baseScreenshot.width,
+              height: baseScreenshot.height,
+              mismatchPixels: comparisonResult.mismatchPixels,
+              mismatchFraction: comparisonResult.mismatchFraction,
+            },
+          ];
+        }
+      )
+    )
+  ).flat();
 
-      comparisonResults.push({
-        baseScreenshotFile,
-        headScreenshotFile,
-        comparisonResult,
-        outcome:
-          comparisonResult.mismatchFraction > diffThreshold ? "fail" : "pass",
-      });
-    }
+  const diffUrl = await getDiffUrl(client, baseReplayId, headReplayId);
+  logger.info(`View screenshot diff at ${diffUrl}`);
 
-    await postScreenshotDiffStats(client, {
+  return [...missingHeadImagesResults, ...headDiffResults];
+};
+
+export const checkScreenshotDiffResult = ({
+  baseReplayId,
+  headReplayId,
+  results,
+  diffOptions,
+}: {
+  baseReplayId: string;
+  headReplayId: string;
+  results: ScreenshotDiffResult[];
+  diffOptions: ScreenshotDiffOptions;
+}): void => {
+  const logger = log.getLogger(METICULOUS_LOGGER_NAME);
+
+  const missingHeadImagesResults = results.filter(
+    ({ outcome }) => outcome === "missing-head"
+  ) as ScreenshotDiffResultMissingHead[];
```

Comment:
[nit -- but worthwhile for future discriminated type union filters since it can occasionally reduce bugs introduced in refactors]

Note if you want to do this type safe, without a cast, you can do:

```ts
const missingHeadImagesResults = results.flatMap(result => result.outcome === "missing-head" ? [result] : []);
```

Typescript will then check the types for you (with current code it won't check that 'ScreenshotDiffResultMissingHead' aligns with the type implied by checking `result.outcome === "missing-head"`).

Reply:
Same with below .filter of course.

Reply:
Done in c07f118